src/grpc/utils.hpp

```diff
@@ -29,4 +29,36 @@ static bool get_file_contents(const std::string& file_name, std::string& content
     return false;
 }
 
-} // namespace sisl::grpc
+[[maybe_unused]] static void serialize_to_byte_buffer(io_blob_list_t const& cli_buf, grpc::ByteBuffer& cli_byte_buf) {
+    folly::small_vector< grpc::Slice, 4 > slices;
+    for (auto const& blob : cli_buf) {
+        slices.emplace_back(blob.cbytes(), blob.size(), grpc::Slice::STATIC_SLICE);
+    }
+    cli_byte_buf.Clear();
+    grpc::ByteBuffer tmp(slices.data(), cli_buf.size());
+    cli_byte_buf.Swap(&tmp);
+}
+
+[[maybe_unused]] static grpc::Status try_deserialize_from_byte_buffer(grpc::ByteBuffer const& cli_byte_buf,
+                                                                      io_blob& cli_buf) {
+    grpc::Slice slice;
+    auto status = cli_byte_buf.TrySingleSlice(&slice);
+    if (status.ok()) {
+        cli_buf.set_bytes(slice.begin());
+        cli_buf.set_size(slice.size());
+    }
+    return status;
+}
+
+[[maybe_unused]] static grpc::Status deserialize_from_byte_buffer(grpc::ByteBuffer const& cli_byte_buf,
```

Comment:
same as above

Reply:
This one is not checked by caller right? that's why we didn't see any error message when we hit this issue?

Reply:
Yes, you are right.